lib/App/DuckPAN/Cmd/New.pm

```diff
@@ -51,33 +157,100 @@ sub run {
 		$lc_name = $lc_path . "_" . $lc_name;
 	}
 
-	$self->app->emit_and_exit(-1, "No templates exist for this IA Type: " . $type->{name}) if (!defined $type->{templates});
+	my @optional_templates = $self->_ask_optional_templates;
 
-	my %template_info = %{$type->{templates}};
-	my $tx            = Text::Xslate->new();
-	my %files         = (
-		test       => ["$filepath.t"],
-		code       => ["$filepath.pm"],
-		handlebars => [$lc_filepath, "$lc_name.handlebars"],
-		js         => [$lc_filepath, "$lc_name.js"]);
 	my %vars = (
 		ia_name           => $name,
 		ia_package_name   => $package_name,
 		ia_name_separated => $separated_name,
 		lia_name          => $lc_name,
-		ia_path           => $filepath
+		ia_path           => $filepath,
+		lia_id            => $lc_name,
```

Comment:
Excellent.  Have the changes to the templates been made as well?

Just FYI, I'll be out for the next two weeks so @moollaza will be trying to move this forward as time permits.